src/yunohost/app.py

```diff
@@ -1092,15 +1078,15 @@ def app_addaccess(operation_logger, apps, users=[]):
     """
     from yunohost.permission import user_permission_update
 
-    permission = user_permission_update(operation_logger, app=apps, permission="main", add_username=users)
+    output = {}
+    for app in apps:
+        permission = user_permission_update(app+".main", add=users, remove="all_users")
+        output[app] = permission["corresponding_users"]
 
-    result = {p : v['main']['allowed_users'] for p, v in permission['permissions'].items()}
+    return {'allowed_users': output}
```

Comment:
Well are we really supporting the old concept of this "depreciated" function ?

As I understand we should add something like that:
if all_users group is allowed -> disallow it

Reply:
I think yes ? c.f. a few lines up there, it `remove=all_users` in all cases, so that should reproduce the old behavior I believe ? Though in the case were `all_users` isn't currently allowed it will display a warning and that's it, but that doesn't sound like a breaking change ?

Reply:
Ah yes didn't see this...